runtime/src/snapshot_utils.rs

```diff
@@ -2942,19 +2942,12 @@ pub fn purge_old_bank_snapshots(
 ) {
     let do_purge = |mut bank_snapshots: Vec<BankSnapshotInfo>| {
```

Comment:
I think `do_purge` as a closure made more sense before some previous changes.

What do you think of just removing this and calling `sort` + `purge_bank_snapshots` below the match? (fine as a follow-up)